lstchain/reco/dl1_to_dl2.py

```diff
@@ -450,17 +450,20 @@ def get_source_dependent_parameters(data, config={}):
 
     src_dep_params = pd.DataFrame(index=data.index)
 
-    is_simu = 'mc_type' in data.columns
+    is_simu = ('mc_type' in data.columns) and ~(data['mc_type'] < 0).all()
     
     if is_simu:
-        if (data['mc_type'] == 0).all():
+        if (data['mc_type'][data['mc_type']>=0] == 0).all():
```

Comment:
Maybe you can include a comment here because this is becoming a bit confusing:
-1 means real data or events that do not pass image cleaning
0 means gamma
>0 hadron

Reply:
ok, I added the comment

Reply:
Why is the information if something passed image cleaning encoded in `mc_type`?

Reply:
Now I understood...!
If events don't pass image cleaning, `mc_type` and `mc_core_distance` is left as default values (-1). But this should be solved by #499 because filling parameters of `mc_type` and `mc_core_distance` is included in a function of `fill_mc`.
I created this PR for the recently processed MC, but seems no need for the future. 
Is it fine to implement this change for the current MC? I guess it will not affect future MC anyway. 
If not, I will withdraw this PR.

Reply:
hi @SeiyaNozaki. This is actually not only affecting MC, but also real data for which `mc_type` will be left as the default -1 as well

Reply:
Yes, exactly. This is because I'd like to implement it. What do you think?